tools/comparables-java/src/main/kotlin/io/spine/tools/comparables/ComparablesJavaRenderer.kt

```diff
@@ -0,0 +1,160 @@
+import io.spine.protodata.codegen.java.JavaRenderer
+import io.spine.protodata.codegen.java.Literal
+import io.spine.protodata.codegen.java.MessageReference
+import io.spine.protodata.codegen.java.MethodCall
+import io.spine.protodata.codegen.java.TypedInsertionPoint.CLASS_SCOPE
+import io.spine.protodata.codegen.java.TypedInsertionPoint.MESSAGE_IMPLEMENTS
+import io.spine.protodata.renderer.SourceSet
+import io.spine.protodata.select
+import java.util.*
+import javax.lang.model.element.Modifier.FINAL
+import javax.lang.model.element.Modifier.PRIVATE
+import javax.lang.model.element.Modifier.PUBLIC
+import javax.lang.model.element.Modifier.STATIC
+import com.squareup.javapoet.ClassName as PoetClassName
+
+/**
+ * A renderer for making messages [Comparable].
+ *
+ * Adds the [Comparable] interface to the message class and generates the `compareTo(...)` method,
+ * which compares the value of the message to another instance of the same message type according
+ * to the rules defined by the [ComparableType] view.
+ *
+ * Note that the [io.spine.tools.comparables.ComparablesPlugin] must be added to ProtoData in order
+ * for this renderer to work.
+ */
+@Suppress("unused") // Used by ProtoData reflectively.
+public class ComparablesJavaRenderer : JavaRenderer() {
+
+    override fun render(sources: SourceSet) {
+        val comparableTypes = select<ComparableType>().all()
+        comparableTypes.forEach { type ->
+            val javaClassName = classNameOf(type.name, type.declaredIn)
+            val canonicalName = javaClassName.canonical
+            val simpleNameIndex = canonicalName.lastIndexOf(".")
+            val packageName = canonicalName.substring(0, simpleNameIndex)
+            val simpleName = canonicalName.substring(simpleNameIndex + 1, canonicalName.length)
+            val selfName = PoetClassName.get(packageName, simpleName)
+            val javaFile = sources.file(javaFileOf(type.name, type.declaredIn))
+            javaFile
+                .at(MESSAGE_IMPLEMENTS.forType(type.name))
+                .add("${Comparable::class.java.canonicalName}<${simpleName}>,")
+            javaFile
+                .at(CLASS_SCOPE.forType(type.name))
+                .withExtraIndentation(level = 1)
+                .add(comparison(selfName, type))
+        }
+    }
+
+    /**
+     * Generates the code for the order comparison.
+     *
+     * The code will contain the overridden `compareTo(..)` as defined in
+     * the [Comparable]interface.
+     */
+    private fun comparison(selfName: PoetClassName, type: ComparableType): List<String> {
+        val other = Literal("other")
+        val naturalOrder = Literal("naturalOrder")
+        val body = CodeBlock
+            .builder()
+            .addStatement("\$T.requireNonNull(\$L)", Objects::class.java, other)
+            .addStatement("return \$L.compare(this, \$L)", naturalOrder, other)
+            .build()
+        val method = MethodSpec
+            .methodBuilder("compareTo")
```

Comment:
Can we have `methodBuilder()` here and `builder()` above coming on the previous line?

Please also consider adding `methodBuilder()` to the list of builder methods in IDEA settings.